src/ucp/core/ucp_mm.c

```diff
@@ -613,26 +636,16 @@ static ucs_status_t ucp_memh_create_from_mem(ucp_context_h context,
         return status;
     }
 
-    if (mem->method != UCT_ALLOC_METHOD_MD) {
-        goto out;
-    }
-
-    for (md_index = 0; md_index < context->num_mds; md_index++) {
-        if (mem->md != context->tl_mds[md_index].md) {
-            continue;
-        }
-
+    md_index = ucp_mem_get_md_index(context, mem->md, mem->method);
+    if (md_index != UCP_NULL_RESOURCE) {
```

Comment:
can we remove the local var md_index and the "out" label?

Reply:
fixed both and retested that original issue is still fixed.